lib/ansible/plugins/connection/ssh.py

```diff
@@ -396,13 +404,25 @@ def _build_command(self, binary, *other_args):
 
         # If we want to use password authentication, we have to set up a pipe to
         # write the password to sshpass.
-
-        if self._play_context.password:
+        pkcs11_provider = C.pkcs11_provider
```

Comment:
this is incorrect, use `self.get_option`

Reply:
I switched to get_option. It wasn't exactly clear how to get at the env/ini variables, the majority of the others seemed to be in the constants object.

Reply:
since 2.4 that is not needed anymore, ssh is a bit overloaded for other plugins so some got left 'as is', but for new options, all you need to do is define them in the plugin's docs.